src/Plugin.php

```diff
@@ -512,8 +514,12 @@ public static function onOpenIdConnectLogin($user, $token_response, $id_token_cl
     if ($last_edit && $last_edit > $user_claims['profile_update_date']) {
       return;
     }
-    update_user_meta($user_id, 'first_name', $user_claims['firstname']);
-    update_user_meta($user_id, 'last_name', $user_claims['lastname']);
+    if (!empty($user_claims['display_firstname'])) {
+      update_user_meta($user_id, 'first_name', $user_claims['display_firstname']);
```

Comment:
Shouldn't we update the values to be empty (remove the values) if neither nor is set to reflect the SSO values? Especially as those fields aren't mandatory fields.

Please see [this](https://app.asana.com/0/633061025239778/890333550594547/f) ticket.